src/Microsoft.DotNet.Darc/src/Darc/Operations/AddBuildToChannelOperation.cs

```diff
@@ -236,10 +238,27 @@ private async Task<bool> ValidateAzDOBuildAsync(AzureDevOpsClient azdoClient, st
         /// <param name="build">Build for which the Arcade SDK dependency build will be inferred.</param>
         private async Task<(string sourceBranch, string sourceVersion)> GetSourceBranchInfoAsync(Build build)
         {
-            if (!string.IsNullOrEmpty(_options.SourceBranch) && !string.IsNullOrEmpty(_options.SourceSHA))
+            var hasSourceBranch = !string.IsNullOrEmpty(_options.SourceBranch);
+            var hasSourceSHA = !string.IsNullOrEmpty(_options.SourceSHA);
+
+            if (hasSourceBranch)
+            {
+                _options.SourceBranch = GitHelpers.NormalizeBranchName(_options.SourceBranch);
+            }
+
+            if (hasSourceBranch && hasSourceSHA)
             {
                 return (_options.SourceBranch, _options.SourceSHA);
             }
+            else if (hasSourceSHA && !hasSourceBranch)
```

Comment:
[nit] this can probably be an early return earlier to improvereadability over the elseif?